python/res/fm/rms/rms_run.py

```diff
@@ -150,10 +151,10 @@ def exec_rms(self, exec_env):
         if self.config.threads:
             args += ["-threads", str(self.config.threads)]
 
-        if exec_env:
-            os.execve(self.config.executable, args, exec_env)
-        else:
-            os.execv(self.config.executable, args)
+        if not exec_env:
+            exec_env = os.environ.copy()
+        exec_env[self.BACKUP_ENVIRONMENT_VARIABLE] = os.environ.get("PYTHONPATH","")
```

Comment:
Why do we need "" in os.environ.get("PYTHONPATH","")?
I tried os.environ.get("PYTHONPATH","") and os.environ.get("PYTHONPATH") and it seemed the same here.

Reply:
`dict.get("SOMETHING")` returns `None` if SOMETHING is not found in the dictionary, while `dict.get("SOMETHING", "")` returns the empty string if it does not find SOMETHING. :slightly_smiling_face:

Reply:
`PYTHONPATH` might be unset, in which case `os.environ.get("PYTHONPATH")` returns `None`. I do question the line here, if `PYTHONPATH` is unset, shouldn't `_PRE_RMS_PYTHONPATH` also be unset?

Reply:
Yes, I also thought about this, but what should the `run_external` script do when it cannot find the `_PRE_RMS_PYTHONPATH` variable? Continue with the current `PYTHONPATH` or unset it?

Reply:
Unset it?

Reply:
I guess you're trying to avoid the scenario where one cannot separate between RMS being run from `rms_run` without  `PYTHONPATH` and just being run not using `rms_run`. That makes a lot of sense, it can potentially be hard to debug, especially since there are already at least two standard ways of running RMS as of today.

However, this information could be backed up by introducing a second environment variable; a `_PRE_RMS_BACKUP` variable that is only set if we backup. In this manner, we can separate the two 🤷